presto-spi/src/main/java/io/prestosql/spi/connector/SchemaRoutineName.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.connector;
+
+import java.util.Objects;
+
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public final class SchemaRoutineName
+{
+    private final String schemaName;
+    private final String routineName;
+
+    public SchemaRoutineName(String schemaName, String routineName)
+    {
+        this.schemaName = requireNonNull(schemaName, "schemaName is null").toLowerCase(ENGLISH);
+        this.routineName = requireNonNull(routineName, "routineName is null").toLowerCase(ENGLISH);
```

Comment:
I don't think lowercasing is appropriate here.

This follows  `SchemaTableName`, but that is about to change
See https://github.com/prestosql/presto/pull/2350/files#diff-d86eb3d71bf172e6ebc672efd6346d4c

Let's get this right from the start.

Reply:
done

Reply:
This change is wrong. We do resolution using lowercase name, but pass mixed case into the access check. So the user can bypass the check by using a mixed case name.